src/-private/changesets/proxy/changeset-proxy-handler-interface.ts

```diff
@@ -0,0 +1,33 @@
+import { Content, PrepareChangesFn, PublicErrors } from '../../../types';
+
+export interface Change {
+  key: string;
+  value: any;
+}
+
+export default interface IChangesetProxyHandler {
+  changes: Record<string, any>[];
+  errors: PublicErrors;
+  error: object;
+  change: object;
+  data: object;
+
+  isChangeset: boolean;
+  isDirty: boolean;
+  isPristine: boolean;
+  isValid: boolean;
+  pendingData: { [index: string]: any };
+  execute(): this;
+  isValidating: (key: string | void) => boolean;
+  prepare(preparedChangedFn: PrepareChangesFn): this;
+  save: (options?: object) => Promise<this | any>;
```

Comment:
related to the v2 proposal and empowering users to use immutability when they see fit, I think we should maybe deprecated save, and move to a `applyChangesTo(passedObject)` method instead. That way folks could opt to never mutate their originating data, and also support various optimistic and pessmistic update flows.

```js
let sourceData = {};

let changeset = createChangeset(sourceData);

changeset.foo = 1;

// whenever you want the sourceData (that maybe happens to be rendered elsewhere) to update:
changeset.applyChangesTo(sourceData); // this is the equiv of .save today, (except we don't call any underlying save method on the sourceData... cause that's a bonkers behavior anyway

// or if you wanted a different object to represent what would be the "next set of changes", like:
let previewChanges = {};
changeset.applyChangesTo(previewChanges);

// and then if you wanted both the original data and a preview of the thanges, you could:

let snapshot = changeset.snapshot();
```